src/main/java/duke/data/DataActions.java

```diff
@@ -0,0 +1,60 @@
+package duke.data;
+
+import duke.output.StandardOutput;
+import duke.output.Symbols;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class DataActions {
+    public static void importData(File dataFile, ArrayList<Task> tasks) {
+        try {
+            Scanner reader = new Scanner(dataFile);
+            while (reader.hasNext()) {
+                String task = reader.nextLine();
+                DataActions.determineTaskType(task, tasks);
+            }
+        } catch (FileNotFoundException e) {
+            System.out.print(StandardOutput.FILE_NOT_FOUND_EXCEPTION_MESSAGE.STANDARD_OUTPUT);
+        }
+    }
+
+    public static void determineTaskType(String task, ArrayList<Task> tasks) {
+        String[] taskInfo = task.split(Symbols.DATA_DELIMITER.SYMBOL); // what if user task name includes " / "
+        if (taskInfo[0].equals(Symbols.TODO.SYMBOL)) {
+            tasks.add(Task.totalTasks, new Todo(taskInfo[2]));
+        } else if (taskInfo[0].equals(Symbols.DEADLINE.SYMBOL)) {
+            // deadline
+            tasks.add(Task.totalTasks, new Deadline(taskInfo[2], taskInfo[3]));
+        } else {
```

Comment:
best to have explicit check for event and use a default branch to deal with invalid command.
now an invalid command could potentially end up being a event.